src/esm.ts

```diff
@@ -147,12 +149,40 @@ export function createEsmHooks(tsNodeService: Service) {
       return defer();
     }
 
-    // pathname is the path to be resolved
+    let candidateSpecifiers: string[] = [specifier];
+
+    if (context.parentURL) {
+      const parentUrl = new URL(context.parentURL);
+      if (parentUrl.pathname && extname(parentUrl.pathname) === '.ts') {
```

Comment:
We'll need to update this to match all parents which should get path mapping, not just `.ts` files.  This potentially includes `tsx`, `mts`, `jsx`, etc.

We can do `if(!tsNodeService.ignored(path)) {...`

Note that will only do mapping if the parent is transformed by us.  For example, given a mix of .ts and .js with allowJs turned off, it will map paths when the parent is .ts but not .js, even if in the same directory.  Should be fine since this PR already excludes .js.

Will also need to handle cases where parentUrl is a non-file URL, just to avoid errors.

Reply:
I’ve updated the code accordingly in https://github.com/TypeStrong/ts-node/pull/1585/commits/48f5262e64adaa270b16a309011857a9722bf2fd